cxx-qt-gen/src/parser/cxxqtdata.rs

```diff
@@ -148,13 +149,18 @@ impl ParsedCxxQtData {
     fn parse_struct(&mut self, s: ItemStruct) -> Result<Option<Item>> {
         // If the attribute is #[cxx_qt::qobject] then this the struct defining a qobject
         if let Some(index) = attribute_find_path(&s.attrs, &["cxx_qt", "qobject"]) {
-            // Remove the macro from the struct
-            //
-            // TODO: we need to read the base class from the macro
-            let mut s = s.clone();
-            s.attrs.remove(index);
-
             if let Some(qobject) = self.qobjects.get_mut(&s.ident) {
+                // Find if there is any base class
+                if let Some(base) = attribute_tokens_to_map::<Ident, LitStr>(&s.attrs[index])?
+                    .get(&quote::format_ident!("base"))
+                {
+                    qobject.base_class = Some(base.value());
+                }
+
+                // Remove the macro from the struct
+                let mut s = s.clone();
```

Comment:
```suggestion
                let mut struct = s.clone();
```
Single letter variable names are a pet peeve of mine.

Reply:
Renaming to `item_struct` as `struct` is reserved.